src/main/java/org/sonar/plugins/stash/issue/collector/StashCollector.java

```diff
@@ -99,150 +102,180 @@ public static StashPullRequest extractPullRequest(String project, String reposit
+          JSONArray jsonLines = (JSONArray)jsonSegment.get("lines");
+
+          if (jsonLines == null) { continue; } // Let's process the next item in "objsegm_loop"
+
+          // Let's call this for loop "objline_loop"
+          for (Object objLine : jsonLines.toArray()) {
+
+            JSONObject jsonLine = (JSONObject)objLine;
+            // destination line in diff view
+            long source = (long)jsonLine.get("source");
+            long destination = (long)jsonLine.get("destination");
+
+            StashDiff diff = new StashDiff(type, path, source, destination);
+            // Add comment attached to the current line
+            JSONArray jsonCommentIds = (JSONArray)jsonLine.get("commentIds");
+
+            if (jsonCommentIds == null) {
+              result.add(diff);
+            } else {
+              // To keep this method depth under control (squid:S134), we outsourced the comments extraction
+              result.add(extractCommentsForDiff(diff, jsonDiff, jsonCommentIds));
             }
           }
         }
       }
-    }
 
+      // Extract File Comments: this kind of comment will be attached to line 0
+      JSONArray jsonLineComments = (JSONArray)jsonDiff.get("fileComments");
+
+      if (jsonLineComments == null) { continue; } // Let's process the next item in "objdiff_loop"
+
+      StashDiff initialDiff = new StashDiff(StashPlugin.CONTEXT_ISSUE_TYPE, path, 0, 0);
+
+      // Let's call this for loop "objlinc_loop"
+      for (Object objLineComment : jsonLineComments.toArray()) {
+
+        JSONObject jsonLineComment = (JSONObject)objLineComment;
+
+        long lineCommentId = (long)jsonLineComment.get("id");
+        String lineCommentMessage = (String)jsonLineComment.get("text");
+        long lineCommentVersion = (long)jsonLineComment.get(VERSION);
+
+        JSONObject objAuthor = (JSONObject)jsonLineComment.get(AUTHOR);
+
+        if (objAuthor == null) { continue; } // Let's process the next item in "objlinc_loop"
+
+        StashUser author = extractUser(objAuthor);
+
+        StashComment comment = new StashComment(lineCommentId, lineCommentMessage, path,
+                                                (long)0, author, lineCommentVersion);
+        initialDiff.addComment(comment);
+      }
+      result.add(initialDiff);
+    }
     return result;
   }
+
+  private static StashDiff extractCommentsForDiff(StashDiff diff, JSONObject jsonDiff, JSONArray jsonCommentIds)
+          throws StashReportExtractionException {
+
+    // Let's call this for loop "objcomm_loop"
+    for (Object objCommentId: jsonCommentIds.toArray()) {
+
+      long commentId = (long)objCommentId;
+      JSONArray jsonLineComments = (JSONArray)jsonDiff.get("lineComments");
+
+      if (jsonLineComments == null) {
+        continue;  // Let's process the next item in "objcomm_loop"
+      }
+
+      // Let's call this for loop "objlico_loop"
+      for (Object objLineComment : jsonLineComments.toArray()) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS135)